omod/src/test/java/org/openmrs/module/sync2/RestResourceConverterTest.java

```diff
@@ -81,5 +77,34 @@ public void convertObject_shouldConvertPersonAttributesToUuids() throws Exceptio
 		assertEquals(1, uuids.size());
 		assertEquals(conceptUuid, uuids.get(0));
 	}
-	
+
+	@Test
+	public void convertObjectShouldConvertEncounter() {
+		SimpleObject encounter = new SimpleObject();
+		final String encounterUuid = "some-encounter-uuid-who-gives-a-damn?";
+		encounter.add("uuid", encounterUuid);
+		SimpleObject obs = createObsSimpleObject();
+		obs.add("encounter", encounter);
+
+		converter.convertObject(WS_REST_V1 + "obs", obs);
+
+		assertEquals(encounterUuid, obs.get("encounter"));
+	}
+
+	@Test
+	public void convertObjectShouldIgnoreIfEncounterNotIncluded() {
+		SimpleObject obs = createObsSimpleObject();
+		converter.convertObject(WS_REST_V1 + "obs", obs);
+		assertFalse(obs.containsKey("encounter"));
```

Comment:
I do not see value in the above assertion since you created obs in the test and you intentionally did not include the encounter key. And converter.convertObject() does not modify `obs`